### PR TITLE
Update change log and update docs in preparation for release

### LINKED ISSUE #3169: Code Coverage Report availability is not documented
Issue #3157 mentions that CI/CD now issues code coverage reports.
This should be documented, maybe in the "Merging Pull Requests" section at 
https://github.com/openenclave/openenclave/blob/master/docs/Community/Contributing.md ?

### PR DESCRIPTION
**Pull request by radhikaj**
1. Update Changelog for v0.10.0
2. Update system ocall optin documentation to be inline with current implementation
3. Document how developers can obtain their code coverage reports. Fixes #3169

### CODE REVIEW

**Comment by radhikaj** [comment]
@yentsanglee Please update with any information I left out for attestation

File: CHANGELOG.md [hunk 1]
```diff
@@ -44,6 +47,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - The Deep-copy feature of oeedger8r is now enabled by default.
 - The oeedger8r-generated header files now contain only the function prototypes. Marshalling structs, function id enums,
   and function tables are generated only in the c files.
+- Docs and scripts updated to use Azure DCAP client v1.6.0.
+- Fix #2930. Fixes the logic of detecting compilers when LVI mitigation is enabled. That is, the old logic always picks clang-7 (if installed) regardless of whether the environment variable CC is set to gcc.
+- Fix #2670. This fix also allows users to specify the version of clang (default is  clang-7) when building the helloworld sample with LVI mitigation.
```
Comment (mingweishih) [inline CHANGELOG.md:52]:
```suggestion
- Fix #2670. This fix also allows users to specify the version of clang (default is clang-7) when building the helloworld sample with LVI mitigation.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -27,13 +27,17 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - The default memory allocator (dlmalloc) can be replaced by providing replacement functions. This ability to plug-in
   a custom allocator is most applicable for multi-threaded enclaves with memory allocation patterns where the default
   memory allocator may not be performant. See [Pluggable Allocators](docs/DesignDocs/Pluggableallocators.md).
+- Gcov is used to obtain code coverage information for the SDK. See [Code Coverage](docs/GettingStartedDocs/Contributors/CodeCoverage.md).
+
```
Comment (yentsanglee) [inline CHANGELOG.md:31]: Please add:

- Added include\openenclave\attestation\attester.h to support attestation plug-in model attester scenarios.
- Added include\openenclave\attestation\verifier.h to support attestation plug-in model verifier scenarios.

File: CHANGELOG.md [hunk 3]
```diff
@@ -44,6 +48,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - The Deep-copy feature of oeedger8r is now enabled by default.
 - The oeedger8r-generated header files now contain only the function prototypes. Marshalling structs, function id enums,
   and function tables are generated only in the c files.
+- Docs and scripts updated to use Azure DCAP client v1.6.0.
+- Fix #2930. Fixes the logic of detecting compilers when LVI mitigation is enabled. That is, the old logic always picks clang-7 (if installed) regardless of whether the environment variable CC is set to gcc.
+- Fix #2670. This fix also allows users to specify the version of clang (default is clang-7) when building the helloworld sample with LVI mitigation.
+- Fix #3056. oe_is_within_enclave() and oe_is_outside_enclave() now reflect the SGX enclave boundary as determined by the enclave SECS rather than the limit of the pages initially provisioned in to the enclave.
+- If not specified, CMAKE_BUILD_TYPE is set to Debug. This ensures that cmake and cmake -DCMAKE_BUILD_TYPE=Debug result in the same build configuration.
```
Comment (yentsanglee) [inline CHANGELOG.md:56]: Please add:

- Moved include/openenclave/attestation/plugin.h to internal. Currently only support internal attestation plugin registration.
- 'flags' parameter is removed from experimental function oe_get_evidence(). Use 'evidence_format' parameter to select evidence format.

**Comment by radhikaj** [comment]
Thanks all.

**Pull request merged**